tests/PHPStan/Analyser/data/asymmetric-properties.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace AsymmetricProperties;
+
+use AllowDynamicProperties;
+use function PHPStan\Testing\assertType;
+
+/**
+ * @property-read int $asymmetricPropertyRw
+ * @property-write int|string $asymmetricPropertyRw
+ *
+ * @property int $asymmetricPropertyXw
+ * @property-write int|string $asymmetricPropertyXw
+ *
+ * @property-read int $asymmetricPropertyRx
+ * @property int|string $asymmetricPropertyRx
+ */
+#[AllowDynamicProperties]
+class Foo
+{
+
+	public function doFoo(): void
+	{
+		assertType('int', $this->asymmetricPropertyRw);
+		assertType('int', $this->asymmetricPropertyXw);
+		assertType('int|string', $this->asymmetricPropertyRx);
```

Comment:
This does not look right. The `@property` should be shadowed by `@property-read`.

Reply:
To me it doesn't make sense to allow both `@property` and `@property-read`. My expectation is that `@property` overwrites everything.

Reply:
Oh, please send an updating PR that restores this behaviour, and adds some tests for it. Thank you!

Reply:
I do not care that much either way – converting my existing `@property` tags to `@property-read` when adding `@property-write` tags is only mildly annoying. Though, if we are counter-intuitively ignoring the more specific tags, we might want to add a check that warns that there are conflicting tags.

Reply:
Because the tags are indexed by name, the check cannot really easily be written.

Reply:
But backward compatibility is important to me so please research it and send a PR.

Reply:
Opened #2329.